src/store/mutations/mutations-model.ts

```diff
@@ -143,6 +143,10 @@ export const mutateCorrectionInformation = (state: StateIF, correctionInformatio
   state.stateModel.correctionInformation = correctionInformation
 }
 
+export const mutateCorrectionFilingDate = (state: StateIF, correctedFilingDate: string) => {
+  state.stateModel.correctionInformation.correctedFilingDate = correctedFilingDate
```

Comment:
The corrected filing's date should not be updated.

You should be updating the correction start date.

https://github.com/bcgov/business-edit-ui/blob/da53affd2acdf9d0123bbf6fde664ee9b12bb6c8/src/interfaces/store-interfaces/state-interfaces/correction-information-interface.ts#L41

Reply:
This actually fixed some errors that I was having with the correction date.

Reply:
I have also done some work on this in my PR as I am updating/fixing the filing template mixin, etc.

Ref: https://github.com/bcgov/business-edit-ui/pull/335/

Reply:
I think this needs to be "mutateCorrectionStartDate". It should be a string (yyyy-mm-dd).

You can get the initial value from a correction's Corrected Filing Date but you'll have to convert it to yyyy-mm-dd. There's a future ticket to provided business start date in the business response and then it will be OK.

Reply:
Okay I will/have changed it to mutateCorrectionStartDate, do I want to get an initial value from correctedFilingDate? Isn't that the date that the filing was made?

Reply:
Good question.

For firm corrections -- where you display Business Start/founding Date -- you can get the initial value from the entity snapshot business object.

For other filing types, it would be best to ask at Standup or with a BA/PO. Moving forward, corrections will correct the current business state (snapshot), not the filing being corrected. (I'm thinking we can obsolete correctedFilingDate at some point, but I think the API looks for it atm.) It boils down to: what should the UI display? The date the corrected filing was filed? The business start/founding date? The recognition date and time (and I'm not sure what this is exactly -- ask).

Reply:
The business response for BENs also include a founding date that looks usable:

![image](https://user-images.githubusercontent.com/10002889/181590854-bfc5e215-3978-488a-bfd3-6c2b06b0873a.png)

Reply:
But I wouldn't display founding date as Recognition Date and Time without checking with PO/BA/UI and perhaps changing the label.

Reply:
When I mutate the correction start date in the same form as this the correction start date does not change. Do I have to do a clone deep to it or something?

Reply:
No need to clone it because it's a primitive type (which is assigned by value), not an object (which is assigned by reference).

I don't think you're saving (storing) the right variable. `correctionInformation.correctedFilingDate` is the filing date of the filing being corrected (an IA or Registration or ChgReg) (although we are getting away from correcting a filing to correcting the current business state).

I think the variable you want to update is `correctionInformation.startDate`. See https://github.com/bcgov/business-edit-ui/blob/de4c94dc420be0d8e387c5384ade030c14056891/src/mixins/filing-template-mixin.ts#L183

Reply:
Resolving this conversation and I will create a more concise (and shorter) one...